include/xsimd/arch/generic/xsimd_generic_memory.hpp

```diff
@@ -55,6 +55,42 @@ namespace xsimd
             return batch<T, A>::load_aligned(concat_buffer);
         }
 
+        // gather
+        namespace detail
+        {
+            template <typename T, typename A, typename U, typename V, size_t I>
+            inline batch<T, A> gather(U const* src, batch<V, A> const& index,
+                                      ::xsimd::detail::index_sequence<I>) noexcept
+            {
+                return insert(batch<T, A> {}, static_cast<T>(src[index.get(I)]),
+                              ::xsimd::index<I>());
+            }
+
+            template <typename T, typename A, typename U, typename V, size_t I,
+                      size_t... Is,
+                      typename = typename std::enable_if<sizeof...(Is) != 0>::type>
```

Comment:
Instead of an `enable_if` here, you should just use `size_t I0, size_t I1, size_t... Is`